src/server_manager/web_app/digitalocean_account.ts

```diff
@@ -0,0 +1,141 @@
+// Copyright 2020 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {DigitalOceanSession} from '../cloud/digitalocean_api';
+import * as crypto from '../infrastructure/crypto';
+import * as do_install_script from '../install_scripts/do_install_script';
+import {Account} from '../model/account';
+import {ManagedServer} from '../model/server';
+
+import {DigitaloceanServer, GetCityId} from './digitalocean_server';
+import {ShadowboxSettings} from './shadowbox_server';
+
+export type RegionId = string;
+export type Location = {
+  id: string; regions: RegionId[];
+};
+
+export enum Status {
+  ACTIVE,
+  EMAIL_UNVERIFIED,
+  INVALID_BILLING_INFORMATION,
+}
+
+export class DigitalOceanAccount implements Account {
+  private static readonly MACHINE_SIZE = 's-1vcpu-1gb';
+  private static readonly SHADOWBOX_TAG = 'shadowbox';
+
+  constructor(
+      private apiClient: DigitalOceanSession, private shadowboxSettings: ShadowboxSettings,
+      private debugMode: boolean, private disconnectFn: () => void) {}
+
+  async getId(): Promise<string> {
+    // TODO: Memoize
+    return (await this.apiClient.getAccount()).uuid;
+  }
+
+  async getName(): Promise<string> {
+    return (await this.apiClient.getAccount()).email;
```

Comment:
Perhaps this to be safer:
```suggestion
    return (await this.apiClient.getAccount())?.email;
```